src/decor/java/com/enderio/decoration/data/model/block/PaintedBlockModelBuilder.java

```diff
@@ -0,0 +1,47 @@
+package com.enderio.decoration.data.model.block;
+
+import com.enderio.EnderIO;
+import com.google.gson.JsonObject;
+import net.minecraft.core.Direction;
+import net.minecraft.world.level.block.*;
+import net.minecraftforge.client.model.generators.CustomLoaderBuilder;
+import net.minecraftforge.client.model.generators.ModelBuilder;
+import net.minecraftforge.common.data.ExistingFileHelper;
+import net.minecraftforge.registries.ForgeRegistries;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Objects;
+
+public class PaintedBlockModelBuilder<T extends ModelBuilder<T>> extends CustomLoaderBuilder<T> {
+
+    private @Nullable Block referenceBlock = null;
+    private Direction itemTextureRotation = Direction.NORTH;
+
+    public static <T extends ModelBuilder<T>> PaintedBlockModelBuilder<T> begin(T parent, ExistingFileHelper existingFileHelper) {
+        return new PaintedBlockModelBuilder<>(parent, existingFileHelper);
+    }
+
+    protected PaintedBlockModelBuilder(T parent, ExistingFileHelper existingFileHelper) {
+        super(EnderIO.loc("painted_block"), parent, existingFileHelper);
+    }
+
+    public PaintedBlockModelBuilder<T> reference(Block referenceBlock) {
+        this.referenceBlock = referenceBlock;
+        return this;
+    }
+
+    public PaintedBlockModelBuilder<T> itemTextureRotation(Direction direction) {
+        this.itemTextureRotation = direction;
+        return this;
+    }
+
+    @Override
+    public JsonObject toJson(JsonObject json) {
+        json = super.toJson(json);
+        json.addProperty("reference", Objects.requireNonNull(ForgeRegistries.BLOCKS.getKey(Objects.requireNonNull(referenceBlock))).toString());
```

Comment:
add message to requireNonNull, so there is a difference in the log between both calls (or remove the nonNull check before the toString on the RL, so that it's a defaulted with string NullPointerException)